### DISCUSSION TITLE
Lack of bass in ES8388

### DISCUSSION THREAD

**Discussion by yujonglee** [pschatzmann/arduino-audiokit#82] [Q&A] [answered]
Hello. I know it sounds little vague, but it seems like sound signature is changed from original source when using `ES8388`.(especially bass really deeps down). I read the datasheet of `ES8388`, looking for default eq or something, but can not find any clues.

I am using `ESP32-LyraT V4.3`, and listen audio output with 3.5mm jack and earphone. 

I tried [this code](https://github.com/espressif/esp-adf/tree/master/examples/player/pipeline_a2dp_sink_stream) from `ESP-ADF` and [this code](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-audiokit/basic-a2dp-audiokit/basic-a2dp-audiokit.ino) from `Arduino-Audiokit`, but got same result.

I am wondering if this is just constraint / performance issue of Codec itself. 

I'd really appreciate it if anyone could share experience and knowledge about what is possible cause of this. Sound quality is important for my application, so if it is codec problem, I need to change it. So recommendation on that will be also very helpful.

**Comment by pschatzmann** [comment, marked as answer]
You did not write if you used pcm audio or some lossy codec.
I never looked into this, but you can look into the  ES8388 shelfing filter settings.

**Reply by yujonglee** [reply]
I played Spotify music in my phone, and send it to `ESP32-LyraT V4.3` using A2DP. I tested with MP3 file in the SD card, but got same result. Please let me know if I am missing any information for you to understand the situation.

Thank you for the pointer to `shelving filter`.  😄 

- In the [datasheet](http://www.everest-semi.com/pdf/ES8388%20DS.pdf), the default value is `{5'h0f, 5'h1f, 5'h0f, 5'h1f, 5'h0f, 5'h1f}`. 
- In the [user guide 24p](https://dl.radxa.com/rock2/docs/hw/ds/ES8388%20user%20Guide.pdf), it said `Everest Semiconductor Co., Ltd will provide equalizer calculator`. I can not yet find the calculator to understand what `{5'h0f, 5'h1f, 5'h0f, 5'h1f, 5'h0f, 5'h1f}` means, but it feels like a really "default" value.

Do you think I need to dig into this and change `ES8388` filter setting to get what I want? Is it natural to adjust codec/DAC to remove some kind of pre-defined bias?

Thank you for the help.
